src/link/http/createHttpLink.ts

```diff
@@ -182,18 +180,26 @@ export const createHttpLink = (linkOptions: HttpOptions = {}) => {
       // removal of window.fetch, which is unlikely but not impossible.
       const currentFetch = preferredFetch || maybe(() => fetch) || backupFetch;
 
+      const observerNext = observer.next.bind(observer);
       currentFetch!(chosenURI, options)
         .then(response => {
           operation.setContext({ response });
           const ctype = response.headers?.get('content-type');
 
           if (ctype !== null && /^multipart\/mixed/i.test(ctype)) {
-            return readMultipartBody(response, observer);
+            return readMultipartBody(response, observerNext);
           } else {
-            return readJsonBody(response, operation, observer);
+            return parseAndCheckHttpResponse(operation)(response).then(observerNext);
           }
         })
-        .catch(err => handleError(err, observer));
+        .then(() => {
+          controller = undefined;
+          observer.complete();
```

Comment:
```suggestion
          observer.complete?.();
```

`parseAndCheckHttpResponse` handled this assuming `complete` wasn't always defined. Do we need the same here?

Reply:
Ugh, it seems that these are actually optional in `zen-observable` - but our tsconfig is so lax that it didn't take that fact into account any more. Good catch!

Reply:
Hah, yes, and no. Our `tsconfig.json` is fine.

This should be typed `SubscriptionObserver` from the start, not `Observer` - and there we're guaranteed that it is present
https://github.com/zenparsing/zen-observable/blob/8406a7e3a3a3faa080ec228b9a743f48021fba8b/src/Observable.js#L163C1-L168C7

Let me fix that type in a few places, that will allow us to remove those checks and save a bit more bundle size.